pkg/flow/memory.go

```diff
@@ -211,30 +210,6 @@ func (im *instanceMemory) StoreData(key string, val interface{}) error {
 	return nil
 }
 
-func (im *instanceMemory) GetAttributes() map[string]string {
-	tags := im.instance.GetAttributes(recipient.Instance)
-	for k, v := range im.tags {
-		tags[k] = v
-	}
-	if im.logic != nil {
-		tags["state-id"] = im.logic.GetID()
-		tags["state-type"] = im.logic.GetType().String()
-	}
-
-	pi := im.engine.InstanceCaller(im)
-	if pi != nil {
-		a := strings.Split(pi.State, ":")
-		if len(a) >= 1 && a[0] != "" {
-			tags["invoker-workflow"] = a[0]
-		}
-		if len(a) > 1 {
-			tags["invoker-state-id"] = a[1]
-		}
-	}
-
-	return tags
-}
-
 func (im *instanceMemory) WithTags(ctx context.Context) context.Context {
 	if im.instance != nil {
 		ctx = im.instance.WithTags(ctx)
```

Comment:
``` go
        if !ok {
		tags = make(map[string]interface{})
	}

	if im.logic != nil {
		tags["state"] = im.logic.GetID()
	}
```


`tags["state"] = im.logic.GetID()` is not safe for concurrent writes

Reply:
i changed to storing the log-attributes directly in the context instead of using a map